### PR TITLE
PDE-476: Handling omitEmptyParams

### PR DESCRIPTION
**Pull request by BrunoBernardino**
Includes tests and has been tested in a CLI app in production with a modified core.

Requires a schema PR: https://github.com/zapier/zapier-platform-schema/pull/57

Fixes [PDE-476](https://zapierorg.atlassian.net/browse/PDE-476).

### CODE REVIEW

File: src/http-middlewares/before/add-query-params.js [hunk 1]
```diff
@@ -2,13 +2,34 @@
...
+
 // Take params off of req.params and append to url - "?a=1&b=2"".
 // This middleware should run *after* custom middlewares, because
 // custom middlewares might add params.
 const addQueryParams = req => {
   if (Object.keys(req.params || {}).length) {
     const splitter = req.url.indexOf('?') === -1 ? '?' : '&';
-    req.url += splitter + querystring.stringify(req.params);
+
+    if (req.omitEmptyParams === true) {
```
Comment (eliangcs) [inline src/http-middlewares/before/add-query-params.js]: Can this just be `if (req.omitEmptyParams)`?

**Review by eliangcs — APPROVED** [review]
The tests look very solid. Thanks for that! Only have a style suggestion (not a blocker). Feel free to merge on your call.

File: src/http-middlewares/before/add-query-params.js [hunk 2]
```diff
@@ -2,13 +2,34 @@
 
 const querystring = require('querystring');
 
+// Mutates the object (it'll be deleted later anyway)
+const removeEmptyParams = params =>
```
Comment (xavdid) [inline src/http-middlewares/before/add-query-params.js:6]: I don’t love that this mutates, but given that it’s local and commented, it’s probably fine

File: src/http-middlewares/before/add-query-params.js [hunk 3]
```diff
@@ -2,13 +2,34 @@
...
-    req.url += splitter + querystring.stringify(req.params);
+
+    if (req.omitEmptyParams === true) {
+      removeEmptyParams(req.params);
+    }
+
+    const stringifiedParams = querystring.stringify(req.params);
+
+    if (stringifiedParams) {
+      req.url += `${splitter}${stringifiedParams}`;
```
Comment (xavdid) [inline src/http-middlewares/before/add-query-params.js:31]: I think this is fine as is, but I think we’d be better off parsing the url with the stdlib, pulling out the query, merging with new params, and adding the whole thing rather than guessing the splitter and adding it that way. 

Again, it’s _fine_, but using established tools feels a little more above the board.

**Review by xavdid — COMMENTED** [review]
Looks great! Love the tests. Had some thoughts about url parsing but nothing showstopping. 

One last thing- do we have tests that cover the `splitter` code? It seems like all of the tests (here and previously) call /get, but none with `/get?a=1`. I may be missing it, but I don’t think anyging uses & as the splitter.

File: src/http-middlewares/before/add-query-params.js (hunk 3 above)
Comment (BrunoBernardino) [reply src/http-middlewares/before/add-query-params.js:31]: Agreed. I just didn't want to change too much here, honestly.

We only test the `splitter` now by not adding `?` (it was in the case the params existed but where then removed).

File: src/http-middlewares/before/add-query-params.js (hunk 2 above)
Comment (BrunoBernardino) [reply src/http-middlewares/before/add-query-params.js:6]: Same here, but since the variable was being mutated/removed before, I went with the less code option here.
